src/utils/verifiable-credential/types.ts

```diff
@@ -0,0 +1,66 @@
+import type { Hex } from '@metamask/utils';
+
+export enum TrustworthinessScope {
+  Honesty = 'Honesty',
+  SoftwareSecurity = 'Software security',
+  SoftwareDevelopment = 'Software development',
+}
+
+export enum SnapCurrentStatus {
+  Endorsed = 'Endorsed',
+  Disputed = 'Disputed',
+}
+
+export enum SnapStatusReasonType {
+  Endorse = 'Endorse',
+  Dispute = 'Dispute',
+}
+
+export enum TrustCredentialType {
+  TrustCredential = 'TrustCredential',
+  VerifiableCredential = 'VerifiableCredential',
+  StatusCredential = 'StatusCredential',
+}
+
+export type PKHDid = `did:pkh:${Hex}`;
+
+export type SnapDid = `snap://${string}`;
+
+export type Did = PKHDid | SnapDid;
+
+export type StatusReason = {
+  type: SnapStatusReasonType;
+  value: string[];
+};
+
+export type Trustworthiness = {
+  scope: TrustworthinessScope;
+  level: number;
+  reason?: string[];
+};
+
+export type AccountCredentialSubject = {
+  id: PKHDid;
+  trustworthiness: Trustworthiness[];
+};
+
+export type SnapCredentialSubject = {
+  id: SnapDid;
+  currentStatus: SnapCurrentStatus;
+  statusReason: StatusReason;
+};
+
+export type CredentialSubject =
+  | AccountCredentialSubject
+  | SnapCredentialSubject;
+
+export type Assertion = {
+  type: TrustCredentialType[];
+  issuer: Did;
+  credentialSubject: CredentialSubject;
+  '@context': string[];
+};
+
+export type SignedAssertion = Assertion & {
+  proof: any;
```

Comment:
we may need to improve the type for proof here in the future to avoid to use `any`, anyway, we can let this pass in this PR.

Reply:
indeed, need to define it more struct after we confirm the proof structure